src/include/detail/linalg/tdb_io.h

```diff
@@ -308,15 +309,21 @@ std::vector<T> read_vector(
   auto dim_num_{domain_.ndim()};
   auto array_rows_{domain_.dimension(0)};
 
-  if (start_pos == 0) {
-    start_pos = array_rows_.template domain<domain_type>().first;
-  }
-  if (end_pos == 0) {
-    end_pos = array_rows_.template domain<domain_type>().second + 1;
+  if (read_full_vector) {
```

Comment:
The API of this function seems a bit inconsistent to me. We already have the notion of if `start_pos, end_pos == 0` we read full vector based on domains. Why do we need an extra `read_full_vector` argument vector for?

The client can either use 0 for all positions or define specific slice to read.

We either need to use domain for size and pass `start_pos, end_pos == 0` or use max domain and maintain the size on the client. Not sure what this will give us on top of that.

Reply:
> We already have the notion of if `start_pos, end_pos == 0` we read full vector based on domains. Why do we need an extra read_full_vector argument vector for?

We need some way of telling this function not to read the full vector. Right now if you pass `start_pos, end_pos == 0` then we read the full vector, but if there is no where to read the vector into, i.e. in this case, then we crash:
```
tiledb::Query query(ctx, *array_);
  query.set_subarray(subarray).set_data_buffer(
      attr_name, data_.data(), vec_rows_);
  tiledb_helpers::submit_query(tdb_func__, uri, query);
```
Does that make sense? This is basically a flag to say "`start_pos, end_pos == 0` means read nothing vs means read everything".

Reply:
Could we have 2 functions?
- with start, end pos
- with just read from domain

Reply:
So there are already two functions, one where you don't pass positions:
```
template <class T>
std::vector<T> read_vector(
    const tiledb::Context& ctx, const std::string& uri, size_t timestamp = 0) {
  return read_vector<T>(ctx, uri, 0, 0, timestamp, true);
}
```
and one where do you:
```
template <class T>
std::vector<T> read_vector(
    const tiledb::Context& ctx,
    const std::string& uri,
    size_t start_pos,
    size_t end_pos,
    size_t timestamp = 0,
    bool read_full_vector = false) {
```
The issue I was seeing is that some callers call into the one with positions with 0 values, and if I don't keep this logic inside the one with positions then we fail tests. But I'l investigate further and try to fix those because you're right, having two separate functions is much simpler.

Reply:
Okay updated to your suggestion, now we just have two functions.